packages/hooks/src/useP2PAdvertList.ts

```diff
@@ -28,9 +29,43 @@ const useP2PAdvertList = (
         }));
     }, [data?.p2p_advert_list?.list, getRate]);
 
+    const [combined_data, setCombinedData] = useState(modified_data);
+    const previous_data_ref = useRef<typeof modified_data>([]);
+
+    useEffect(() => {
+        previous_data_ref.current = [];
+        invalidate('p2p_advert_list');
+    }, [
+        payload?.advertiser_id,
+        payload?.counterparty_type,
+        payload?.payment_method,
+        payload?.use_client_limits,
+        payload?.local_currency,
+        payload?.sort_by,
+        invalidate,
+    ]);
+
+    useEffect(() => {
+        // This preserves the previous data when loadMore is called.
```

Comment:
@jim-deriv You shouldn't need these things 🤔 `keepPreviousData` inside the `usePaginatedFetch` should do this for you out of the box 🤔

Reply:
@farzin-deriv, as I checked this wasn't happening

Reply:
@jim-deriv It might be something is wrong inside `usePaginatedFetch`, Or somewhere else 🤔 

Please have a look at their example and see if you can find what we did wrong 🙏🏻 Thanks man 🌹 
https://tanstack.com/query/v4/docs/react/examples/react/pagination

Reply:
Sure thing @farzin-deriv